infer/src/base/DBWriter.ml

```diff
@@ -420,14 +441,27 @@ end
 
 let use_daemon =
   let is_windows = match Version.build_platform with Windows -> true | Linux | Darwin -> false in
-  Config.((not is_windows) && dbwriter && (not (buck || genrule_mode)) && jobs > 1)
+  let do_use = Config.((not is_windows) && dbwriter && (not (buck || genrule_mode)) && jobs > 1) in
+  if do_use then
+    (* Only the main process should try detecting whether the socket can be created.
+       Otherwise, re-spawn Infer will try to create a socket on top of the existing one. *)
+    if Config.is_originator then
+      if Server.can_use_socket () then true
+      else (
+        L.user_warning
+          "Cannot setup the socket to communicate with the database daemon. Performance will be \
+           impacted. Do you have enough rights to create a Unix socket in directory '%s'?@."
+          Config.toplevel_results_dir ;
+        false )
+    else Server.socket_exists ()
+  else false
 
 
 let perform cmd = if use_daemon then Server.send cmd else Command.execute cmd
 
 let start () = Server.start ()
 
-let stop () = Server.send Command.Terminate
+let stop () = try Server.send Command.Terminate with Unix.Unix_error _ -> ()
```

Comment:
Is there a separate issue that the socket creation could succeed but this could fail? As far as I can tell this is only called when `use_daemon` is true.

Reply:
I'm not 100% sure anymore. To be honest, I wondered about that too when I prepared the code. I will have another look.

Reply:
Ah actually, I have a nice commit message about exactly this:
```
    Intercept error when shutting down db daemon
    
    Results in much nicer behavior when Infer is interrupted.
```
I will experiment with soft and hard kill and report on the differences with and without this change.

Reply:
If it's a pain to check we can just leave this in.

Reply:
Without this patch, this is why I currently get when I Ctrl-C a capture that uses DBWriter:
```
465395: Error while running epilogue "Stop Sqlite write daemon":
Unix.Unix_error(Unix.ENOENT, "connect", "").
Powering through...
```
This may be due to our patches for Ada, but I don't think so.
Note that I would be fine with limiting the catch-all to those arguments if you think this helps.